e2e/specs/hostframe.spec.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describe("Host communication", () => {
+    before(() => {
+        cy.visit("hostframe.html");
+        cy.get("#toolbar button").should("have.length", 9);
+        cy.get("iframe")
+            .iframe(() => {
+                cy.waitForScriptFinish()
+            });
+    });
+
+    it("handles a host theme message", () => {
+        cy.get("iframe")
+            .iframe(() => {
+                cy.prepForElementSnapshots()
+            });
+
+        cy.get("iframe").first().matchImageSnapshot("theme-message-before");
+        cy.getIndexed("#toolbar > button", 0).click();
```

Comment:
nit (here and throughout this file): Is there a way to get these elements via a text query so that this is a bit more human-readable?